### ISSUE TITLE
Problem with latest version (on Fedora 28)

### ISSUE THREAD

**Issue by Fale** [Spotifyd/spotifyd#138]
Hi,
I upgraded from Fedora 27 to Fedora 28 and it broke my configuration.
For this reason, I decided to:

* download the latest version in git (https://github.com/Spotifyd/spotifyd/commit/2c13ff926931ff9d341b3918d2c244cf4fbf65d5)
* compile it with: `cargo build --release --features pulseaudio_backend,dbus_mpris`
* run it with `target/release/spotifyd --no-daemon`

But it fails with:
```
10:08:53 [INFO] Using software volume controller.
10:08:53 [INFO] Connecting to AP "gew1-accesspoint-b-jd98.ap.spotify.com:4070"
10:08:53 [INFO] Authenticated as "USERNAMEHERE" !
10:08:53 [INFO] Country: "IT"
10:08:53 [ERROR] Caught panic with message: called `Result::unwrap()` on an `Err` value: WireError(InvalidEnumValue(14))
```

My config file is `~/.config/spotifyd/spotifyd.conf` and contains:
```
[global]
username = USERNAMEHERE
password = PASSWORDHERE
backend = pulseaudio
#onevent = command_run_on_playback_event
device_name = x250.fale.io
bitrate = 320
cache_path = /tmp/spotify
volume-normalisation = true
#normalisation-pregain = -10
```

**Comment by Fale** [comment]
Adding the log in verbose mode:

```
10:16:22 [TRACE] mio::poll: [<unknown>:785] registering with poller
10:16:22 [TRACE] tokio_threadpool::builder: [<unknown>:326] build; num-workers=4
10:16:22 [DEBUG] tokio_reactor::background: starting background reactor
10:16:22 [INFO] Using software volume controller.
10:16:22 [DEBUG] librespot_connect::discovery: Zeroconf server listening on 0.0.0.0:0
10:16:22 [TRACE] mio::poll: [<unknown>:785] registering with poller
10:16:22 [TRACE] mio::poll: [<unknown>:785] registering with poller
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Writable Token(4194305)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [TRACE] mio::poll: [<unknown>:785] registering with poller
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Writable Token(8388610)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [TRACE] mio::poll: [<unknown>:785] registering with poller
10:16:22 [TRACE] hyper::client::pool: [<unknown>:125] park; waiting for idle connection: "http://apresolve.spotify.com"
10:16:22 [TRACE] hyper::client::connect: [<unknown>:118] Http::connect("http://apresolve.spotify.com/")
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Writable Token(12582915)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [DEBUG] tokio_core::reactor: consuming notification queue
10:16:22 [DEBUG] hyper::client::dns: resolving host="apresolve.spotify.com", port=80
10:16:22 [TRACE] mdns::fsm: [/home/fale/.cargo/git/checkouts/rust-mdns-881ed19b93df0e9d/733b2b6/src/fsm.rs:275] sending packet to V4(224.0.0.251:5353)
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(4194305)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [TRACE] mdns::fsm: [/home/fale/.cargo/git/checkouts/rust-mdns-881ed19b93df0e9d/733b2b6/src/fsm.rs:275] sending packet to V6([ff02::fb]:5353)
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(8388610)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(8388610)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 1817501 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 776533321 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 103190 }
10:16:22 [TRACE] mdns::fsm: [/home/fale/.cargo/git/checkouts/rust-mdns-881ed19b93df0e9d/733b2b6/src/fsm.rs:76] received packet from V4(192.168.56.123:5353)
10:16:22 [TRACE] mdns::fsm: [/home/fale/.cargo/git/checkouts/rust-mdns-881ed19b93df0e9d/733b2b6/src/fsm.rs:87] received packet from V4(192.168.56.123:5353) with no query
10:16:22 [TRACE] mdns::fsm: [/home/fale/.cargo/git/checkouts/rust-mdns-881ed19b93df0e9d/733b2b6/src/fsm.rs:76] received packet from V6([fe80::c49b:c277:ea4:62db]:5353)
10:16:22 [TRACE] mdns::fsm: [/home/fale/.cargo/git/checkouts/rust-mdns-881ed19b93df0e9d/733b2b6/src/fsm.rs:87] received packet from V6([fe80::c49b:c277:ea4:62db]:5353) with no query
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 194118 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 776857128 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 34742 }
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 14580 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 776923967 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 37284 }
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Writable Token(4194305)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 1086172 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 778062229 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 52824 }
10:16:22 [DEBUG] hyper::client::connect: connecting to 104.199.64.136:80
10:16:22 [TRACE] mio::poll: [<unknown>:785] registering with poller
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Writable Token(16777220)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 30021739 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 808484644 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 97868 }
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:311] read_keep_alive; is_mid_message=false
10:16:22 [TRACE] hyper::proto: [<unknown>:122] should_keep_alive(version=Http11, header=None) = true
10:16:22 [TRACE] hyper::proto::h1::role: [<unknown>:349] Client::encode has_body=false, method=None
10:16:22 [TRACE] hyper::proto::h1::io: [<unknown>:546] reclaiming write buf Vec
10:16:22 [DEBUG] hyper::proto::h1::io: flushed 47 bytes
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:627] flushed State { reading: Init, writing: KeepAlive, keep_alive: Busy, error: None, read_task: None }
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 490422 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 809310601 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 23113 }
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(16777220)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:187] Conn::read_head
10:16:22 [DEBUG] hyper::proto::h1::io: read 686 bytes
10:16:22 [TRACE] hyper::proto::h1::role: [<unknown>:254] Response.parse([Header; 100], [u8; 686])
10:16:22 [TRACE] hyper::proto::h1::role: [<unknown>:259] Response.parse Complete(268)
10:16:22 [TRACE] hyper::header: [<unknown>:355] maybe_literal not found, copying "Keep-Alive"
10:16:22 [TRACE] hyper::header: [<unknown>:355] maybe_literal not found, copying "Vary"
10:16:22 [DEBUG] hyper::proto::h1::io: parsed 9 headers (268 bytes)
10:16:22 [DEBUG] hyper::proto::h1::conn: incoming body is content-length (418 bytes)
10:16:22 [TRACE] hyper::proto: [<unknown>:133] expecting_continue(version=Http11, header=None) = false
10:16:22 [TRACE] hyper::proto: [<unknown>:122] should_keep_alive(version=Http11, header=Some(Connection([KeepAlive]))) = true
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:273] Conn::read_body
10:16:22 [TRACE] hyper::proto::h1::decode: [<unknown>:88] decode; state=Length(418)
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:627] flushed State { reading: Body(Length(0)), writing: KeepAlive, keep_alive: Busy, error: None, read_task: None }
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 30788270 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 840133383 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 24744 }
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:273] Conn::read_body
10:16:22 [TRACE] hyper::proto::h1::decode: [<unknown>:88] decode; state=Length(0)
10:16:22 [DEBUG] hyper::proto::h1::conn: incoming body completed
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:440] maybe_notify; read_from_io blocked
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:311] read_keep_alive; is_mid_message=false
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:627] flushed State { reading: Init, writing: Init, keep_alive: Idle, error: None, read_task: None }
10:16:22 [TRACE] hyper::client::pool: [<unknown>:301] pool dropped, dropping pooled ("http://apresolve.spotify.com")
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 375893 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 840764450 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 39937 }
10:16:22 [INFO] Connecting to AP "gew1-accesspoint-b-v3pk.ap.spotify.com:4070"
10:16:22 [TRACE] mio::poll: [<unknown>:785] registering with poller
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:311] read_keep_alive; is_mid_message=false
10:16:22 [TRACE] hyper::proto::h1::dispatch: [<unknown>:400] client tx closed
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:873] State::close_read()
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:880] State::close_write()
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:627] flushed State { reading: Closed, writing: Closed, keep_alive: Disabled, error: None, read_task: None }
10:16:22 [TRACE] hyper::proto::h1::conn: [<unknown>:635] shut down IO
10:16:22 [TRACE] mio::poll: [<unknown>:905] deregistering handle with poller
10:16:22 [DEBUG] tokio_reactor: dropping I/O source: 4
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 191891 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 895182400 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 21591 }
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 2549 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 895216785 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 18092 }
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Writable Token(20971525)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 43137948 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 938381789 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 72940 }
10:16:22 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(20971525)
10:16:22 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:22 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 29579987 }
10:16:22 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2831, tv_nsec: 980095079 }
10:16:22 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 71322 }
10:16:22 [TRACE] tokio_io::framed_write: [<unknown>:182] flushing framed transport
10:16:22 [TRACE] tokio_io::framed_write: [<unknown>:185] writing; remaining=107
10:16:22 [TRACE] tokio_io::framed_write: [<unknown>:202] framed transport flushed
10:16:23 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(20971525)
10:16:23 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 124622078 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 116362991 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 84373 }
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [INFO] Authenticated as "USERNAMEHERE" !
10:16:23 [DEBUG] librespot_core::session: new Session[0]
10:16:23 [DEBUG] librespot_connect::spirc: new Spirc[0]
10:16:23 [DEBUG] librespot::component: new MercuryManager
10:16:23 [DEBUG] librespot_playback::player: new Player[0]
10:16:23 [DEBUG] librespot_playback::audio_backend::pulseaudio: Using PulseAudio sink
10:16:23 [DEBUG] librespot_connect::spirc: input volume:65535 to mixer: 65535
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [DEBUG] librespot_core::session: Session[0] strong=4 weak=2
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [INFO] Country: "IT"
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:182] flushing framed transport
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:185] writing; remaining=832
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:202] framed transport flushed
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 523773 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 117517802 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 48228 }
10:16:23 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(20971525)
10:16:23 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:182] flushing framed transport
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:202] framed transport flushed
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 212835 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 117805700 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 43754 }
10:16:23 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(20971525)
10:16:23 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:182] flushing framed transport
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:202] framed transport flushed
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 2398216 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 120269339 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 42918 }
10:16:23 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(20971525)
10:16:23 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:182] flushing framed transport
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:202] framed transport flushed
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 1339014 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 121673504 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 31148 }
10:16:23 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(20971525)
10:16:23 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:182] flushing framed transport
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:202] framed transport flushed
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 1069811 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 122791226 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 42257 }
10:16:23 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(20971525)
10:16:23 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:182] flushing framed transport
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:202] framed transport flushed
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 29755121 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 152611510 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 58726 }
10:16:23 [DEBUG] librespot_core::mercury: subscribed uri=hm://remote/3/user/USERNAMEHERE/ count=0
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 8224 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 152842414 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 39478 }
10:16:23 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(20971525)
10:16:23 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:182] flushing framed transport
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:202] framed transport flushed
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 12006726 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 164904132 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 27742 }
10:16:23 [DEBUG] dbus_tokio::adriver: Modify_watch: Watch { fd: 11, read: true, write: false }, poll_now: false
10:16:23 [TRACE] mio::poll: [<unknown>:785] registering with poller
10:16:23 [DEBUG] dbus_tokio::adriver: Dropping AConnection
10:16:23 [DEBUG] dbus_tokio::adriver: D-Bus i/o poll ready: 11 is NotReady
10:16:23 [DEBUG] dbus_tokio::adriver: handle_msgs: (Signal, Some("/org/freedesktop/DBus"), Some("org.freedesktop.DBus"), Some("NameAcquired"))
10:16:23 [DEBUG] dbus_tokio::adriver: handle_msgs: (Signal, Some("/org/freedesktop/DBus"), Some("org.freedesktop.DBus"), Some("NameAcquired"))
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 66785 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 166545898 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 23615 }
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 4080 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 166584896 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 20096 }
10:16:23 [TRACE] tokio_reactor: [<unknown>:345] event Readable | Writable Token(20971525)
10:16:23 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:192] frame decoded from buffer
10:16:23 [TRACE] tokio_io::framed_read: [<unknown>:189] attempting to decode a frame
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:182] flushing framed transport
10:16:23 [TRACE] tokio_io::framed_write: [<unknown>:202] framed transport flushed
10:16:23 [DEBUG] tokio_core::reactor: loop poll - Duration { secs: 0, nanos: 27443347 }
10:16:23 [DEBUG] tokio_core::reactor: loop time - Instant { tv_sec: 2832, tv_nsec: 194058871 }
10:16:23 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 32874 }
10:16:23 [DEBUG] dbus_tokio::adriver: Polling message stream
10:16:23 [DEBUG] dbus_tokio::adriver: msgstream found Ok(Ready(Some((Signal, Some("/org/freedesktop/DBus"), Some("org.freedesktop.DBus"), Some("NameAcquired")))))
10:16:23 [WARN] Unhandled dbus message: (Signal, Some("/org/freedesktop/DBus"), Some("org.freedesktop.DBus"), Some("NameAcquired"))
10:16:23 [DEBUG] dbus_tokio::adriver: Polling message stream
10:16:23 [DEBUG] dbus_tokio::adriver: msgstream found Ok(Ready(Some((Signal, Some("/org/freedesktop/DBus"), Some("org.freedesktop.DBus"), Some("NameAcquired")))))
10:16:23 [WARN] Unhandled dbus message: (Signal, Some("/org/freedesktop/DBus"), Some("org.freedesktop.DBus"), Some("NameAcquired"))
10:16:23 [DEBUG] dbus_tokio::adriver: Polling message stream
10:16:23 [DEBUG] dbus_tokio::adriver: msgstream found Ok(NotReady)
10:16:23 [ERROR] Caught panic with message: called `Result::unwrap()` on an `Err` value: WireError(InvalidEnumValue(14))
10:16:23 [DEBUG] librespot_connect::spirc: drop Spirc[0]
10:16:23 [DEBUG] librespot_playback::player: Shutting down player thread ...
10:16:23 [DEBUG] librespot_playback::player: drop Player[0]
10:16:23 [DEBUG] librespot_core::session: drop Session[0]
10:16:23 [DEBUG] librespot::component: drop MercuryManager
10:16:23 [DEBUG] dbus_tokio::adriver: Dropping AMessageStream
10:16:23 [DEBUG] dbus_tokio::adriver: AMessageStream telling ADriver to quit
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:64] shutdown; state=State { lifecycle: 0, num_futures: 0 }
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:110]   -> transitioned to shutdown
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:122]   -> shutting down workers
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:126]   -> shutdown worker; idx=3; state=WorkerState { lifecycle: "WORKER_SHUTDOWN", is_pushed: true }
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:141] signal_stop -- WORKER_SHUTDOWN; idx=3
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:176] worker_terminated; num_workers=3
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:126]   -> shutdown worker; idx=2; state=WorkerState { lifecycle: "WORKER_SHUTDOWN", is_pushed: true }
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:141] signal_stop -- WORKER_SHUTDOWN; idx=2
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:176] worker_terminated; num_workers=2
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:126]   -> shutdown worker; idx=1; state=WorkerState { lifecycle: "WORKER_SHUTDOWN", is_pushed: true }
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:141] signal_stop -- WORKER_SHUTDOWN; idx=1
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:176] worker_terminated; num_workers=1
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:126]   -> shutdown worker; idx=0; state=WorkerState { lifecycle: "WORKER_SHUTDOWN", is_pushed: true }
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:141] signal_stop -- WORKER_SHUTDOWN; idx=0
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:176] worker_terminated; num_workers=0
10:16:23 [TRACE] tokio_threadpool::inner: [<unknown>:179] notifying shutdown task
10:16:23 [TRACE] tokio_threadpool::shutdown: [<unknown>:38] Shutdown::poll
10:16:23 [TRACE] tokio_reactor: [<unknown>:345] event Readable Token(4194303)
10:16:23 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
10:16:23 [DEBUG] tokio_reactor::background: shutting background reactor down NOW
10:16:23 [DEBUG] tokio_reactor::background: background reactor has shutdown
10:16:23 [DEBUG] librespot_core::session: drop Dispatch
```

**Comment by hlbstd** [comment]
Same here

**Comment by SimonPersson** [comment]
This issue is fixed by #135. Thanks for reporting.
